tests/unit/test_client.py

```
@@ -0,0 +1,357 @@
+        result = network_client.fetch_url_with_retry("https://example.com", max_retries=3)
+        assert mock_get.call_count == 3
+        
+    def test_core_client_concurrent_processing(self, core_client, sample_html):
+        """Test concurrent HTML processing."""
+        import threading
+        results = []
+        def process_html():
+            result = core_client.process_html(sample_html)
+            results.append(result)
+        threads = [threading.Thread(target=process_html) for _ in range(5)]
+        for thread in threads: thread.start()
```

Ruff (0.11.9)

Multiple statements on one line (colon)

(E701)